api/proto/teleport/legacy/types/types.proto

```diff
@@ -385,6 +385,37 @@ message DatabaseStatusV3 {
   ];
 }
 
+// IAMPolicyStatus represents states that describe if an AWS database
+// has its IAM policy properly configured or not.
+// This enum is set in a Sync.Map during an IAM task that checks for the
+// validity of IAM policy, and the database gets updated with the value
+// from this map during a heartbeat.
+enum IAMPolicyStatus {
```

Comment:
Can we use a simple string with values like `success`/`failure` instead of a enum? The "problem" with enums is that while technically they are more appropriate as "sum types", they're represented as numbers so less human readable than just a string. So it's much harder to reason about the value of the field if you see e.g. `policy_status: 1` vs `policy_status: success`.

So I would just make a simple string and use a  "fake" enum in Go code like we do in many other places to represent types e.g.

```go
type PolicyStatus string

PolicyStatusSuccess PolicyStatus = "success"
PolicyStatusFailure PolicyStatus = "failure"
...
```

Reply:
@Joerger for this file [tsh_test.go](https://github.com/gravitational/teleport/pull/29721/files#diff-6e9a47fd2352fc235c71625260ad632836576115c91ede28eda4c483e003a6e2R3673), when i tested it with an enum, it printed a numerical value (0 for unspecified). is there a way to print it as `IAM_POLICY_STATUS_UNSPECIFIED`?

Reply:
Ah you're right, you'd need to add customized json marshaling instead:

```go
func (s *IAMPolicyStatus) MarshalJSON() ([]byte, error) {
	return json.Marshal(s.String())
}

func (s *IAMPolicyStatus) UnmarshalJSON(data []byte) error {
	if len(data) == 0 {
		return nil
	}

	var stringVar string
	if err := json.Unmarshal(data, &stringVar); err != nil {
		return err
	}

	*s = IAMPolicyStatus(IAMPolicyStatus_value[stringVar])
	return nil
}
```